Extras/FrontendTests/tests/basic_stream.spec.ts

```diff
@@ -10,20 +11,22 @@ test('Test default stream.', {
 }, async ({ page, streamerId }) => {
 
     await page.goto(`/?StreamerId=${streamerId}`);
-    await page.getByText('Click to start').click();
 
     // let the stream run for a short duration
-    await helpers.waitForVideo(page);
-    await helpers.delay(1000);
-
-    // query the frontend for its calculated stats
-    const frame_count:number = await page.evaluate(()=> {
-        let videoStats = pixelStreaming._webRtcController.peerConnectionController.aggregatedStats.inboundVideoStats;
-        return videoStats.framesReceived;
+    await helpers.startAndWaitForVideo(page);
+
+    let frameCount: number = await page.evaluate(()=> {
+        return new Promise<number>((resolve) => {
+            window.pixelStreaming.addEventListener("statsReceived", (e: StatsReceivedEvent) => {
+                if(e.data.aggregatedStats && e.data.aggregatedStats.inboundVideoStats && e.data.aggregatedStats.inboundVideoStats.framesReceived) {
+                    resolve(e.data.aggregatedStats.inboundVideoStats.framesReceived);
+                }
+            });
+        });
```

Comment:
Change this to be callback based instead of asking for stats straight away to fix test flakiness where it would ask for stats rarely before there was any